Logger/BaseLogger.cs

```diff
@@ -1,7 +1,20 @@
-﻿namespace Logger
-{
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace Logger { 
+
     public abstract class BaseLogger
     {
-        public abstract void Log(LogLevel logLevel, string message);
+        public abstract void Log(LogLevel? logLevel, string message);
+
+        public string? ClassName { get; set; } = string.Empty;
```

Comment:
```suggestion
        public string? ClassName { get; set; }
```
No need to set this to an empty string. It can be left as null.